core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -198,9 +198,18 @@ UNWIND(	.cantunwind)
 #if defined(CFG_HWSUPP_MEM_PERM_WXN) && defined(CFG_CORE_RWDATA_NOEXEC)
 	orr	r0, r0, #(SCTLR_WXN | SCTLR_UWXN)
 #endif
+#if defined(CFG_SCTLR_ROUNDROBIN_REPLACEMENT)
+	orr	r0, r0, #SCTLR_RR
+#endif
 	write_sctlr r0
 	isb
 
+#if defined(CFG_EARLY_BRANCH_PREDICTION)
+	write_bpiall
```

Comment:
these sync are not used when flushing btac in other places (cache, tlb).
Should it be fixed also ?